File: src/network/packet/impl/ConsoleLogPacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl;

use pocketcloud\cloud\console\log\CloudLogger;
use pocketcloud\cloud\network\client\ServerClient;
use pocketcloud\cloud\network\packet\ClientboundPacket;
use pocketcloud\cloud\network\packet\CloudboundPacket;
use pocketcloud\cloud\network\packet\CloudPacket;
use pocketcloud\cloud\network\packet\data\LogType;
use pocketcloud\cloud\network\packet\util\PacketData;

final class ConsoleLogPacket extends CloudPacket implements ClientboundPacket, CloudboundPacket {

    public function __construct(
        private string $message = "",
        private ?LogType $logType = null
    ) {}

    public function handle(ServerClient $client): void {
        CloudLogger::get()->log($this->logType->toLogLevel(), $this->message);
    }

    public function encodePayload(PacketData $packetData): void {
        $packetData->writeAll($this->message, $this->logType);
    }

    public function decodePayload(PacketData $packetData): void {
        $packetData->readAllTypeSafe([&$this->message, &$this->logType], [fn() => $packetData->readString(), fn() => $packetData->readLogType()]);
    }

    public function getMessage(): string {
        return $this->message;
    }

    public function getLogType(): ?LogType {
        return $this->logType;
    }

    public static function create(string $message, LogType $logType): self {
        return new self($message, $logType);
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/network/packet/impl/ConsoleLogPacket.java
package de.pocketcloud.cloud.network.packet.impl;

import de.pocketcloud.cloud.network.client.ServerClient;
import de.pocketcloud.network.packet.AuthenticatedPacket;
import de.pocketcloud.network.packet.ClientboundPacket;
import de.pocketcloud.network.packet.CloudboundPacket;
import de.pocketcloud.cloud.network.packet.CloudPacket;
import de.pocketcloud.cloud.network.packet.type.LogType;
import de.pocketcloud.network.packet.data.PacketData;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

@NoArgsConstructor
@Getter
public final class ConsoleLogPacket extends CloudPacket implements ClientboundPacket, CloudboundPacket, AuthenticatedPacket {

    private String message;
    private LogType logType;

    public ConsoleLogPacket(String message, LogType logType) {
        this.message = message != null ? message : "";
        this.logType = logType;
    }

    @Override
    public void handle(@NotNull ServerClient client) {
        client.server().logger().log(logType.toLogLevel(), message);
    }

    @Override
    public void encodePayload(PacketData packetData) {
        packetData.writeAll(message, logType);
    }

    @Override
    public void decodePayload(PacketData packetData) {
        this.message = packetData.readString();
        this.logType = packetData.readEnum(LogType.class);
    }

    public static ConsoleLogPacket create(String message, LogType logType) {
        return new ConsoleLogPacket(message, logType);
    }
}
